src/variety/interbd.js

```diff
@@ -0,0 +1,720 @@
+
+		key_interbd: function(ca) {
+			var cell = this.cursor.getc();
+			var color = -1;
+
+			for (var row in this.keyrows) {
+				color = this.keyrows[row].indexOf(ca);
+				if (color !== -1) {
+					color += 1;
+					break;
+				}
+			}
+
+			if (color !== -1) {
+				if (cell.ques !== color || cell.qnum === -1) {
+					cell.setQues(color);
+				} else {
+					cell.setQues(0);
+					color = -1;
+				}
+
+				if (color === -1 && cell.qnum === -2) {
+					cell.setQnum(-1);
+				} else if (cell.qnum === -1) {
+					cell.setNum(-2);
+				}
+				cell.draw();
+			} else {
+				this.key_inputqnum(ca);
+			}
+		}
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		// qnum: Number clue. -2 is question mark, -1 is empty, 0..4 is regular clue.
+		// ques: Color clue. 0 for no color, 1..6 for clue. When value is non-zero, qnum cannot be empty.
+		// qans: Answer. 0 for empty, 1 for black cell.
+		// qsub: Mark. 0 for empty, 1 for dot.
+
+		numberRemainsUnshaded: true,
+
+		minnum: 0,
+		maxnum: 4,
+
+		getNumColor: function() {
+			if (this.qnum === -1 && this.ques === 0) {
+				return -1;
+			}
+			var num = this.qnum < 0 ? 0 : this.qnum + 1;
+			return this.ques * 10 + num;
+		},
+
+		setNumColor: function(val) {
+			if (val < 0) {
+				this.setQnum(-1);
+				this.setQues(0);
+			} else {
+				var num = val % 10;
+				this.setQnum(num > 0 ? num - 1 : -2);
+				this.setQues((val / 10) | 0);
+				this.setQans(0);
+				this.setQsub(0);
+			}
+		},
+
+		posthook: {
+			qnum: function(num) {
+				if (num === -1) {
+					this.setQues(0);
+				}
+			}
+		}
+	},
+	AreaUnshadeGraph: {
+		enabled: true
+	},
+
+	//---------------------------------------------------------
+	// 画像表示系
+	Graphic: {
+		numbercolor_func: "qnum",
+		qanscolor: "black",
+		circleratio: [0.43, 0.37],
+
+		colors: ["gray", "red", "blue", "green", "#c000c0", "#ff8000", "#00c0c0"],
+
+		paint: function() {
+			this.drawBGCells();
+			this.drawShadedCells();
+			this.drawDotCells();
+
+			this.drawGrid();
+			this.drawCircles();
+
+			this.drawQuesMarks();
+			this.drawQuesNumbers();
+
+			this.drawChassis();
+
+			this.drawTarget();
+		},
+
+		getCircleStrokeColor: function(cell) {
+			if (this.puzzle.getConfig("disptype_interbd") !== 1) {
+				return null;
+			}
+
+			if (cell.ques < 1 || cell.ques > 6) {
+				return cell.qnum !== -1 ? this.colors[0] : null;
+			}
+			return this.colors[cell.ques];
+		},
+
+		getQuesNumberText: function(cell) {
+			if (
+				cell.qnum === -2 &&
+				(this.puzzle.getConfig("disptype_interbd") === 1 || cell.ques !== 0)
+			) {
+				return "";
+			}
+
+			return this.getNumberText(cell, cell.qnum);
+		},
+
+		getQuesNumberColor: function(cell) {
+			if (this.puzzle.getConfig("disptype_interbd") === 2) {
+				return this.quescolor;
+			}
+			return this.colors[cell.ques];
+		},
+
+		getNumberVerticalOffset: function(cell) {
+			this.fontsizeratio =
+				this.puzzle.getConfig("disptype_interbd") === 1 || cell.ques === 0
```

Comment:
```suggestion
				this.puzzle.getConfig("disptype_interbd") === 1
```

I think the "shape" modes look nicer when the unshaped clues have the same font size (so they don't look like a different thing).

Reply:
What do you think of removing the circle from the list of "colors", and using that as "no shape assigned" instead? (And probably do something to the style to make it less heavy -- gray, or a thinner line, or dashed even if we can.)

Reply:
Here's what that could look like (hacky change, lost the highest color): 
<img width="410" alt="Screenshot 2020-10-26 at 13 14 59" src="https://user-images.githubusercontent.com/132113/97171178-4cb9ba80-178d-11eb-9a01-c47af7586823.png">

Reply:
For monochrome puzzles, Mellowmelon uses squares for clues without a color, and circles/diamonds for red and blue regions. Some of his other puzzles go up to 6 colors, and I was running out of discernable shapes when designing the colorblind mode, so I went with using no shape for the clues with unknown regions instead.

I dislike the idea of giving one of the shapes a special meaning, since it would be harder to explain and reason with (e.g. "All different shapes correspond to a region, except for circles which must be replaced with one of the other shapes"). I don't feel as strongly about the grey clues, since it's easier to explain that grey means "unknown color".

Reply:
Random thoughts:
- the distinction could be "circle" vs. "polygon" instead of "specific shape" vs. "all the other shapes"
- while the distinction between "gray circle" and "black polygon" is strong enough for my taste in that screenshot, maybe there's something else we could do? e.g. draw a tighter thin circle? or have the uncolored clues black text on a gray disc (filled circle)?
- as an alternative (or past 6), could add a small letter in the top left corner of a cell?

Reply:
Here's a mellowmelon monochrome puzzle btw: https://mellowmelon.wordpress.com/2011/02/16/puzzle-284/

This uses a thinner outline on the "uncolored" clues btw. And there's a comment critical of this presentation. :)

Reply:
That's the puzzle I was thinking of, altough I missed that comment when I implemented this puzzle. I have to say I agree with it completely 😛

Reply:
Here's how it would look with solid grey circles:
![afbeelding](https://user-images.githubusercontent.com/7021451/97204288-3542f780-17b6-11eb-846d-627c404bae60.png)

Reply:
Hmm that doesn't really look nice, does it?

Here's two takes on the thinner & smaller circle (gray and black).

<img width="321" alt="Screenshot 2020-10-26 at 20 16 03" src="https://user-images.githubusercontent.com/132113/97217955-4ac11d00-17c8-11eb-898f-95fcecb2e49e.png">
<img width="321" alt="Screenshot 2020-10-26 at 20 16 03" src="https://user-images.githubusercontent.com/132113/97217973-4e54a400-17c8-11eb-8003-f9ebe632edad.png">

Reply:
My slight preference right now is for that version I just shared (thin smaller circles, gray text), but if you prefer naked numbers, I'm happy to merge that, too.

And I'd suggest switching to the "both" representation as a general default.

One other vague thought I had is that the triangles aren't a particularly pretty shape, so perhaps to put them later in the rotation?

Reply:
Does it look better if the triangles point up instead of down?

![afbeelding](https://user-images.githubusercontent.com/7021451/97219860-e94e7d80-17ca-11eb-8968-b90006593df2.png)

Reply:
I think I prefer having no shape on the numbers over using the thin circle. I might color the numbers black in Shapes/Both mode instead of grey, so they stand out more.

Reply:
I think pointing up is an improvement, yeah.